testsuite/cluster-test/src/prometheus.rs

```diff
@@ -100,9 +102,45 @@ impl Prometheus {
             .avg()
             .ok_or_else(|| format_err!("Failed to compute avg"))
     }
+
+    pub fn query_range_max(
+        &self,
+        query: String,
+        start: &Duration,
+        end: &Duration,
+        step: u64,
+    ) -> Result<HashMap<String, f64>> {
+        let response = self.query_range(query, start, end, step)?;
+        response
+            .max()
+            .ok_or_else(|| format_err!("Failed to compute max"))
+    }
 }
 
 impl MatrixResponse {
+    pub fn max(&self) -> Option<HashMap<String, f64>> {
+        if self.inner.is_empty() {
+            return None;
+        }
+        let mut max = HashMap::new();
+        for (peer_id, time_series) in &self.inner {
+            let cur_max = time_series
+                .inner
+                .iter()
+                .map(|tuple| tuple.1)
+                // Some time series can return NaN, but f64::max always
+                // chooses the non-NaN argument
+                .fold(f64::NAN, f64::max);
+
+            max.insert(peer_id.clone(), cur_max);
+        }
```

Comment:
FYI: since `HashMap` implements `FromIterator`, you can also:
```
        let max: HashMap<_, _> = self
            .inner
            .iter()
            .map(|(peer_id, time_series)| {
                (
                    peer_id.clone(),
                    time_series
                        .inner
                        .iter()
                        .map(|tuple| tuple.1)
                        // Some time series can return NaN, but f64::max always
                        // chooses the non-NaN argument
                        .fold(0. / 0., f64::max),
                )
            })
            .collect();
```
but that's just a matter of taste, and what you have is perfectly suitable.